### PR TITLE
Add assert to make Draggables are used properly

### PR DESCRIPTION
**Pull request by luanpotter**
Analogous with Tapable

### CODE REVIEW

File: lib/src/game/base_game.dart [hunk 1]
```diff
@@ -47,6 +43,12 @@ class BaseGame extends Game with FPSCounter {
         'Tapable Components can only be added to a BaseGame with HasTapableComponents',
       );
     }
+    if (c is Draggable) {
```
Comment (erickzanardo) [inline lib/src/game/base_game.dart:46]: I think that we could add a test for this, and for the others as well, what do you think?

Comment (erickzanardo) [reply lib/src/game/base_game.dart:46]: 🧠 🧙‍♂️

File: lib/src/game/base_game.dart [hunk 2]
```diff
@@ -2,15 +2,11 @@ import 'dart:ui';
...
 import 'package:ordered_set/comparing.dart';
 import 'package:ordered_set/ordered_set.dart';
 
-import '../components/component.dart';
-import '../components/mixins/has_game_ref.dart';
-import '../components/mixins/tapable.dart';
-import '../components/position_component.dart';
-import '../extensions/vector2.dart';
+import '../../components.dart';
+import '../../extensions.dart';
```
Comment (erickzanardo) [inline lib/src/game/base_game.dart:9]: Do we know if that affects tree shaking?

**Review by erickzanardo — APPROVED** [review]
One small concern, but LGTM already

**Review by wolfenrain — APPROVED** [review]

File: lib/src/game/base_game.dart (hunk 2 above)
Comment (luanpotter) [reply lib/src/game/base_game.dart:9]: I will revert for now until we have renan's answer

Comment (luanpotter) [reply lib/src/game/base_game.dart:9]: cc @renancaraujo

**Pull request merged**